src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -568,6 +571,31 @@ public Boolean[] executeSimple(Object[][] params) {
 		}).description("Returns true if the input is NaN (not a number).")
 			.examples("isNaN(0) # false", "isNaN(0/0) # true", "isNaN(sqrt(-1)) # true")
 			.since("2.8.0");
+
+		Functions.registerFunction(new SimpleJavaFunction<String>("formatNumber", new Parameter[] {
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("format", DefaultClasses.STRING, true, new SimpleLiteral<String>(DEFAULT_NUMBER_FORMAT.toPattern(), true))
+		}, DefaultClasses.STRING, true) {
+			@Override
+			public String[] executeSimple(Object[][] params) {
+				Number number = (Number) params[0][0];
+				String pattern = (String) params[1][0];
+
+				try {
+					return CollectionUtils.array(new DecimalFormat(pattern).format(number));
```

Comment:
I think we could actually reuse the DecimalFormat instance for each thread, via ThreadLocal#withInitial and the DecimalFormat#applyPattern method